### PR TITLE
Mqtt chatroom

### PR DESCRIPTION
**Pull request by vomov**

### CODE REVIEW

File: cheatsheet.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+```
+
+```bash
+nano Dockerfile
+```
+
+`ENV id 1054`
+
+```bash
+nano pom.xml
```
Comment (mtjon) [inline cheatsheet.md:112]: This is unacceptable.

File: cheatsheet.md [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+```
+
+```xml
+<version>1.0.1-SNAPSHOT-MARCO</version>
```
Comment (mtjon) [inline cheatsheet.md]: might want to choose a more generic name

Comment (mtjon) [reply cheatsheet.md]: on second thought... the name doesn't get more generic than this.

File: cheatsheet.md [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+mvn package
+```
+
+```bash
+docker login dataserviceshub-docker-$TENANT.jfrog.io -p $jfrog_password -u $jfrog_login
+```
+
+```bash
+docker push \
+  dataserviceshub-docker-$TENANT.jfrog.io/tenant-example:1.0.1-SNAPSHOT-MARCO
```
Comment (mtjon) [inline cheatsheet.md]: see earlier comment on generic naming

File: cheatsheet.md [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+
+## OR Simplified
+
+```bash
+docker login $jfrog_url -p $jfrog_password -u $jfrog_login
+```
+
+```bash
+docker push \
+  $jfrog_url/tenant-example:1.0.1-SNAPSHOT-MARCO
```
Comment (mtjon) [inline cheatsheet.md]: generic naming consistency

File: cheatsheet.md [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+```bash
+echo $ump_api_endpoint
+echo $API_KEY
+```
+
+## marco snapshot
+
+```json
+{
+  "image": "dataserviceshub-docker-training.jfrog.io/tenant-example:1.0.1-SNAPSHOT-MARCO",
```
Comment (mtjon) [inline cheatsheet.md]: generic naming

File: cheatsheet.md [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+ ```
+
+## UMP
+
+```bash
+echo $ump_api_endpoint
+echo $API_KEY
+```
+
+## marco snapshot
```
Comment (mtjon) [inline cheatsheet.md]: generic naming.

File: mqtt.md [hunk 7]
```diff
@@ -202,9 +201,9 @@ export THING_ID=...
 
 ```bash
 curl -X POST \
-"https://api.$PLATFORM.kpn-dsh.com/datastreams/v0/mqtt/token" \
+"https://api."$PLATFORM".kpn-dsh.com/datastreams/v0/mqtt/token" \
```
Comment (mtjon) [inline mqtt.md]: These extra quotes are unncessary, see: http://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Single-Quotes

**Review by mtjon — CHANGES_REQUESTED** [review]
Please make sure naming in the cheatsheet is generic and does not refer to a single trainer (marco in this case).
Please change back the variable substitution within double quotes (`"`) as referenced in the Bash Manual.

File: mqtt.md (hunk 7 above)
Comment (vomov) [reply mqtt.md]: Fixed

File: cheatsheet.md (hunk 6 above)
Comment (vomov) [reply cheatsheet.md]: Fixed

File: cheatsheet.md (hunk 5 above)
Comment (vomov) [reply cheatsheet.md]: Fixed

File: cheatsheet.md (hunk 4 above)
Comment (vomov) [reply cheatsheet.md]: Fixed

File: cheatsheet.md (hunk 3 above)
Comment (vomov) [reply cheatsheet.md]: Fixed

File: cheatsheet.md (hunk 2 above)
Comment (vomov) [reply cheatsheet.md]: Managed to make more generic by removing the generic name

File: cheatsheet.md (hunk 1 above)
Comment (vomov) [reply cheatsheet.md:112]: added Vim

Comment (mtjon) [reply cheatsheet.md:112]: ;)

File: mqtt.md [hunk 8]
```diff
@@ -204,7 +203,7 @@ export THING_ID=...
 curl -X POST \
 "https://api.$PLATFORM.kpn-dsh.com/datastreams/v0/mqtt/token" \
 -H "Authorization: Bearer `cat rest-token.txt`" \
--d '{"id":"'$THING_ID'"}' > mqtt-token.txt
+-d '{"id":"'$THING_ID'"}' > mqtt-token_1.txt
```
Comment (mtjon) [inline mqtt.md]: Isn't this too much of a hint? Let them struggle.

**Review by mtjon — CHANGES_REQUESTED** [review]

**Review by mtjon — APPROVED** [review]
LGTM

**Pull request merged**